pkg/chunk/chunk_store.go

```diff
@@ -154,6 +160,14 @@ func (c *Store) Get(ctx context.Context, from, through model.Time, allMatchers .
 	if through < from {
 		return nil, fmt.Errorf("invalid query, through < from (%d < %d)", through, from)
 	}
+	if c.cfg.MaxQueryRangeDays == 0 {
+		// Specified as unlimited
+	} else {
+		if through.After(from.Add(time.Hour * 24 * time.Duration(c.cfg.MaxQueryRangeDays))) {
```

Comment:
you will just need to use `c.cfg.MaxQueryRangeDays` as the value after switching to `f.DurationVar`.